src/t_set.c

```diff
@@ -1067,6 +1068,14 @@ void sunionDiffGenericCommand(client *c, robj **setkeys, int setnum,
             return;
         }
         sets[j] = setobj;
+        if (op == SET_OP_DIFF) {
+            if (!sets[0]) {
+                break;
+            } else if (j > 0 && sets[0] == sets[j]) {
+                sameset = 1; 
+                break;
+            }
+        }
```

Comment:
```suggestion
        if (j > 0 && sets[0] == sets[j]) {
            sameset = 1; 
        }
```

I don't think we can afford to break the loop. it means we skip the future `checkType` checks, and could return a diff result instead of a WRONGTYPE error.

p.s. the fact we skip processing the set members is enough IMHO.. no need to try and skip the loop over the input keys.